File: php/renderers/tilesetRenderer.php
<?php

class TilesetRenderer
{
	protected $tileset = null;

	public function __construct($tileset)
	{
		$this->tileset = $tileset;
	}

	public function render($format = 'png')
	{
		$t = $this->tileset;

		$imageX		= 288;
		$imageY		= ($t['count'] / 9) * 32;
		$offsetX	= 0;
		$offsetY	= 0;
		
		$im = imagecreatetruecolor($imageX, $imageY);

		for($i = 0; $i < $t['count']; $i++)
		{
			$this->renderTile($im, $i, $offsetX, $offsetY);

			$offsetX += 32;
			if(($offsetX + 32) > $imageX)
			{
				$offsetX	=  0;
				$offsetY	+= 32;
			}
		}

		ob_start();
		switch($format)
		{
			case 'jpg':
			case 'jpeg':
				imagejpeg($im);
			break;

			case 'gif':
				imagegif($im);
			break;

			case 'png':
			default:
				imagepng($im);
			break;
		}

		$image = ob_get_contents();
		ob_end_clean();

		return $image;
	}

	public function renderTile($im, $tileIndex, $offsetX, $offsetY)
	{
		if(!isset($this->tileset['tiles'][$tileIndex]))
		{
			return;
		}
		
		$tile = $this->tileset['tiles'][$tileIndex];

		$x = 0;
		$y = 0;
		foreach($tile as $xData)
		{
			foreach($xData as $yData)
			{
				$color = imagecolorallocate($im, $yData[0], $yData[1], $yData[2]);
				imagesetpixel($im, $offsetX + $x, $offsetY + $y, $color);

				$y++;
			}

			$x++;
			$y = 0;
		}
	}
}

File: python/renderers/tilesetRenderer.py
# coding=UTF-8
import pygame, sys

tileSurfaces = dict()

def render(tileset):
	windowX = 288
	windowY = (tileset['count']/9)*32

	pygame.init()
	window = pygame.display.set_mode((windowX, windowY)) 
	pygame.display.set_caption(tileset['filename'] + ' - ' + str(tileset['count']) + ' tiles') 
	screen = pygame.display.get_surface() 
	offsetX = 0
	offsetY = 0

	for tileIndex in range(tileset['count']):
		renderTile(screen, tileset, tileIndex, offsetX, offsetY)

		offsetX += 32

		if (offsetX + 32) > windowX:
			offsetX = 0
			offsetY += 32

	pygame.display.update()

	while True:
		for event in pygame.event.get(): 
			if event.type == pygame.QUIT: 
				sys.exit(0)

def renderTile(surface, tileset, tileIndex, offsetX, offsetY):
	global tileSurfaces

	if tileset['filename'] not in tileSurfaces:
		tileSurfaces[tileset['filename']] = dict()

	if tileIndex not in tileSurfaces[tileset['filename']]:
		thisTileSurface = pygame.Surface((32, 32))
		tile = tileset['tiles'][tileIndex]
		pixel = 0

		pixelArray = pygame.PixelArray(thisTileSurface)
		x = 0
		y = 0
		for xData in tile:
			for yData in xData:
				pixelArray[x, y] = yData
				y += 1
			x += 1
			y = 0

		tileSurfaces[tileset['filename']][tileIndex] = pixelArray.make_surface()


	surface.blit(tileSurfaces[tileset['filename']][tileIndex], (offsetX, offsetY))